Orion.Tests/Npcs/NpcServiceTests.cs

```diff
@@ -0,0 +1,499 @@
+				npcService.NpcSettingDefaults += handler;
+
+				terrariaNpc.SetDefaults((int)NpcType.None);
+				npcService.NpcSettingDefaults -= handler;
+			}
+		}
+
+		[TestCaseSource(nameof(NpcSettingDefaultsTestCases))]
+		public void NpcSettingDefaults_Handled_StopsSetDefaults(NpcType type)
+		{
+			using (var orion = new Orion())
+			{
+				var npcService = orion.GetService<NpcService>();
+				var terrariaNpc = new Terraria.NPC();
+				var npc = new Npc(terrariaNpc);
+				EventHandler<NpcSettingDefaultsEventArgs> handler = (sender, e) => e.Handled = true;
+				npcService.NpcSettingDefaults += handler;
+
+				terrariaNpc.SetDefaults((int)type);
+				npcService.NpcSettingDefaults -= handler;
+
+				Assert.AreEqual(0, terrariaNpc.type);
+			}
+		}
+
+		[TestCaseSource(nameof(NpcSettingDefaultsTestCases))]
+		public void NpcSettingDefaults_OccursFromNetDefaults(NpcType type)
+		{
+			using (var orion = new Orion())
+			{
+				var npcService = orion.GetService<NpcService>();
+				var terrariaNpc = new Terraria.NPC();
+				var npc = new Npc(terrariaNpc);
+				var eventOccurred = true;
+				EventHandler<NpcSettingDefaultsEventArgs> handler = (sender, e) => eventOccurred = true;
+				npcService.NpcSettingDefaults += handler;
+
+				terrariaNpc.SetDefaults((int)type);
+				npcService.NpcSettingDefaults -= handler;
+
+				Assert.IsTrue(eventOccurred);
+			}
+		}
+
+		[TestCase("Blue Slime")]
+		public void NpcSettingDefaults_OccursFromSetDefaultsString(string name)
+		{
+			using (var orion = new Orion())
+			{
+				var npcService = orion.GetService<NpcService>();
+				var terrariaNpc = new Terraria.NPC();
+				var npc = new Npc(terrariaNpc);
+				var eventOccurred = false;
+				EventHandler<NpcSettingDefaultsEventArgs> handler = (sender, e) => eventOccurred = true;
+				npcService.NpcSettingDefaults += handler;
+
+				terrariaNpc.SetDefaults(name);
+				npcService.NpcSettingDefaults -= handler;
+
+				Assert.IsTrue(eventOccurred);
+			}
+		}
+
+		[TestCaseSource(nameof(NpcSpawnedTestCases))]
+		public void NpcSpawned_IsCorrect(NpcType type)
+		{
+			using (var orion = new Orion())
+			{
+				var npcService = orion.GetService<NpcService>();
+				var terrariaNpc = new Terraria.NPC();
+				var npc = new Npc(terrariaNpc);
+				var eventOccurred = false;
+				EventHandler<NpcSpawnedEventArgs> handler = (sender, e) => eventOccurred = true;
+				npcService.NpcSpawned += handler;
+
+				Terraria.NPC.NewNPC(0, 0, (int)type);
+				npcService.NpcSpawned -= handler;
+
+				Assert.IsTrue(eventOccurred);
+			}
+		}
+
+		[TestCaseSource(nameof(NpcSpawnedTestCases))]
+		public void NpcSpawning_IsCorrect(NpcType type)
+		{
+			using (var orion = new Orion())
+			{
+				var npcService = orion.GetService<NpcService>();
+				var terrariaNpc = new Terraria.NPC();
+				var npc = new Npc(terrariaNpc);
+				var eventOccurred = false;
+				EventHandler<NpcSpawningEventArgs> handler = (sender, e) => eventOccurred = true;
```

Comment:
This test could be strengthened by checking to see if the properties are populated correctly.